ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/BlockFetch/ClientInterface.hs

```diff
@@ -188,7 +189,8 @@ mkBlockFetchConsensusInterface
     blockMatchesHeader = Block.blockMatchesHeader
 
     readCandidateChains :: STM m (Map peer (AnchoredFragment (Header blk)))
-    readCandidateChains = getCandidates
+    readCandidateChains = undefined getCandidates
+       -- REVIWE: Should we change the 'BlockFetchConsensusInterface' s that 'readCandidateChains' return 'HeaderWithTime'?
```

Comment:
I see. This would change the type of `readCurrentChain`, which I'm unsure it is intended (since now the current chain will need to incorporate the times).

Reply:
The candidates are now AFs of header with time, and the current chain remains an AF of headers. This was enabled by https://github.com/IntersectMBO/ouroboros-network/pull/5010